mock-certify-plugin/src/main/java/io/mosip/certify/mock/integration/mocks/MdocGenerator.kt

```diff
@@ -0,0 +1,121 @@
+package io.mosip.certify.mock.integration.mocks
+
+import co.nstant.`in`.cbor.CborBuilder
+import co.nstant.`in`.cbor.CborEncoder
+import co.nstant.`in`.cbor.model.DataItem
+import com.android.identity.credential.NameSpacedData
+import com.android.identity.internal.Util
+import com.android.identity.mdoc.mso.MobileSecurityObjectGenerator
+import com.android.identity.mdoc.util.MdocUtil
+import com.android.identity.util.Timestamp
+import io.mosip.certify.util.*
+import java.io.ByteArrayOutputStream
+import io.mosip.certify.util.IssuerKeyPairAndCertificate
+import java.util.*
+
+
+class MdocGenerator {
+    companion object {
+        const val NAMESPACE: String = "org.iso.18013.5.1"
+        const val DOCTYPE: String = "$NAMESPACE.mDL"
+        const val DIGEST_ALGORITHM = "SHA-256"
+        const val ECDSA_ALGORITHM = "SHA256withECDSA"
+        const val SEED = 42L
+    }
+
+    fun generate(
+        data: MutableMap<String, out Any>,
+        holderId: String,
+        p12FileName: String,
+        password: String,
+        caAlias: String,
+        issuerAlias: String
+    ): String? {
+        val issuerKeyPairAndCertificate: IssuerKeyPairAndCertificate? = readKeypairAndCertificates(
+            p12FileName, password, listOf(caAlias,issuerAlias)
+        )
+        if(issuerKeyPairAndCertificate == null) {
+            throw RuntimeException("Unable to load Crypto details")
+        }
+        val devicePublicKey = JwkToKeyConverter().convertToPublicKey(holderId.replace("did:jwk:", ""))
+        val issuerKeypair = issuerKeyPairAndCertificate.issuerKeypair()
+
+        val nameSpacedDataBuilder: NameSpacedData.Builder = NameSpacedData.Builder()
+        data.keys.forEach { key ->
+            nameSpacedDataBuilder.putEntryString(NAMESPACE, key, data[key].toString())
+        }
+        val nameSpacedData: NameSpacedData =
+            nameSpacedDataBuilder
+                .build()
+        val generatedIssuerNameSpaces: MutableMap<String, MutableList<ByteArray>> =
+            MdocUtil.generateIssuerNameSpaces(nameSpacedData, Random(SEED), 16)
+        val calculateDigestsForNameSpace =
+            MdocUtil.calculateDigestsForNameSpace(NAMESPACE, generatedIssuerNameSpaces, DIGEST_ALGORITHM)
+
+        val mobileSecurityObjectGenerator = MobileSecurityObjectGenerator(DIGEST_ALGORITHM, NAMESPACE, devicePublicKey)
+        mobileSecurityObjectGenerator.addDigestIdsForNamespace(NAMESPACE, calculateDigestsForNameSpace)
+        val distantFuture: Long = kotlinx.datetime.Instant.Companion.DISTANT_FUTURE.toEpochMilliseconds()
```

Comment:
Is this variable used to specify expiry time of the credential? Can you rename this to `expirationTime`. I believe this is the time of this VC's validity.

Reply:
Resolving the conversation as its modified